### DISCUSSION TITLE
Button press out of cache.

### DISCUSSION THREAD

**Discussion by ManuelRoyBo** [Pycord-Development/pycord#1029] [Q&A] [answered]
Hello! I am working with buttons. Is there a way I could respond to button presses that are not in cache?
I use `button.callback = function` for buttons in cache but ofc when I restart my bot (and clear the cache) it doesn't work.
Is there a way I could get the interaction of a button that is pressed? thx
(Had a "on_message" like event on my mind)

**Comment by krittick** [comment, marked as answer]
Have you tried using persistent views for the first question? https://github.com/Pycord-Development/pycord/blob/master/examples/views/persistent.py

For the second, you can use the `on_interaction` event, or (preferably) just use the `interaction` variable declared in the button callback. I'd also recommend either using the button decorator or subclassing Button instead of directly assigning to the callback as it gives you more flexibility. E.g.  https://github.com/Pycord-Development/pycord/blob/master/examples/views/confirm.py
